client/my-sites/checkout/checkout/emergent-paywall-box.jsx

```diff
@@ -0,0 +1,246 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import page from 'page';
+import classNames from 'classnames';
+import { get, isEqual, debounce, startsWith, isBoolean } from 'lodash';
+import { connect } from 'react-redux';
+import { localize, translate } from 'i18n-calypso';
+import debug from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import analytics from 'lib/analytics';
+import notices from 'notices';
+import TermsOfService from './terms-of-service';
+import wp from 'lib/wp';
+import { paymentMethodName, paymentMethodClassName } from 'lib/cart-values';
+import { getCurrentUserCountryCode } from 'state/current-user/selectors';
+
+const wpcom = wp.undocumented();
+const log = debug( 'calypso:checkout:payment:emergent-payall' );
+
+export class EmergentPaywallBox extends Component {
+	static propTypes = {
+		cart: PropTypes.object.isRequired,
+		selectedSite: PropTypes.object,
+		transaction: PropTypes.object.isRequired,
+	};
+
+	static defaultProps = {
+		selectedSite: {},
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = this.getInitialState();
+		this.iframeRef = React.createRef();
+		this.formRef = React.createRef();
+		this.fetchIframeConfiguration = debounce( this.fetchIframeConfiguration, 500 );
+	}
+
+	componentDidMount() {
+		this.fetchIframeConfiguration();
+		window.addEventListener( 'message', this.onMessageReceiveHandler, false );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'message', this.onMessageReceiveHandler );
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( ! isEqual( prevProps.cart.total_cost, this.props.cart.total_cost ) ) {
+			this.fetchIframeConfiguration();
+		}
+	}
+
+	getInitialState() {
+		return {
+			paywall_url: '',
+			paymentMethod: paymentMethodClassName( 'emergent-paywall' ),
+			payload: '',
+			charge_id: '',
+			signature: '',
+			iframeHeight: 600,
+			iframeWidth: 750,
+			hasConfigLoaded: false,
+			redirectTo: '',
+			pendingOrder: false,
+			siteSlug: this.props.selectedSite ? this.props.selectedSite.slug : 'no-site',
+		};
+	}
+
+	/**
+	 * Determines what to do after a `PURCHASE_STATUS` message
+	 *
+	 * @param {Object} purchaseStatus - the status
+	 * @return {Function?} whichever action is triggered
+	 */
+	handlePurchaseStatusMessage( purchaseStatus ) {
+		if ( isBoolean( purchaseStatus.submitted ) ) {
+			log( 'Setting state.pendingOrder to the prepareOrder promise' );
+			return this.setState( { pendingOrder: this.prepareOrder() } );
+		}
+
+		if ( isBoolean( purchaseStatus.success ) ) {
```

Comment:
this `isBoolean` check is redundant because `true === purchaseStatus.success` tests both that it's `true` and also that it's `boolean`